pkg/review/job.go

```diff
@@ -0,0 +1,141 @@
+// Copyright 2024 The Authors (see AUTHORS file)
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package review
+
+import (
+	"context"
+	"fmt"
+	"runtime"
+
+	"github.com/abcxyz/github-metrics-aggregator/pkg/bq"
+	"github.com/abcxyz/github-metrics-aggregator/pkg/version"
+	"github.com/abcxyz/pkg/githubauth"
+	"github.com/abcxyz/pkg/logging"
+	"github.com/abcxyz/pkg/workerpool"
+)
+
+// ExecuteJob runs the pipeline job to read GitHub
+// commits to check if they were properly reviewed.
+func ExecuteJob(ctx context.Context, cfg *Config) error {
```

Comment:
What happens if this job fails halfway though?

Reply:
Status of each commit is stored at the end, if the job fails in the middle it will not write updated status and it will start over on next run. This is the same functionality as the original Beam implementation.

Reply:
Okay cool. I just want to make sure we don't (more easily) end up in a half-baked state because of differences between Beam and Cloud Run Jobs.